include/openthread/thread.h

```diff
@@ -931,6 +931,23 @@ const otMleCounters *otThreadGetMleCounters(otInstance *aInstance);
  */
 void otThreadResetMleCounters(otInstance *aInstance);
 
+/**
+ * Gets the number of seconds since the device last attached.
+ *
+ * Requires the `OPENTHREAD_CONFIG_UPTIME_ENABLE` feature.
+ *
+ * If the device is not currently attached, zero will be returned.
+ *
+ * Unlike the role-tracking variables in `otMleCounters`, which track the cumulative time the device is in each role,
+ * this function tracks the time since the last successful attachment, indicating how long the device has been
+ * connected to the Thread mesh (regardless of its role, whether acting as a child, router, or leader).
+ *
+ * @param[in] aInstance  A pointer to an OpenThread instance.
+ *
+ * @returns The number of seconds since last attached.
+ */
+uint32_t otThreadGetSecondsSinceLastAttached(otInstance *aInstance);
```

Comment:
What about something like `otThreadGetCurrentAttachDurationSeconds()`? This helps make clear that the value is only valid while the device is attached.

Reply:
Good suggestion. Thanks. I renamed it in the new push. I used `GetCurrentAttachDuration()` (not adding `Seconds` at the end, since we can add the unit in the documentation).